pkg/imgpkg/bundle/bundle.go

```diff
@@ -29,24 +31,83 @@ type Bundle struct {
 	plainImg         *plainimg.PlainImage
 	imgRetriever     ctlimg.ImagesMetadata
 	imagesLockReader ImagesLockReader
+	imagesRef        map[string]ImageRef
 }
 
 func NewBundle(ref string, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
 	return NewBundleWithReader(ref, imagesMetadata, &singleLayerReader{})
 }
 
 func NewBundleFromPlainImage(plainImg *plainimg.PlainImage, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
-	return &Bundle{plainImg, imagesMetadata, &singleLayerReader{}}
+	return &Bundle{plainImg: plainImg, imgRetriever: imagesMetadata, imagesLockReader: &singleLayerReader{}, imagesRef: map[string]ImageRef{}}
 }
 
 func NewBundleWithReader(ref string, imagesMetadata ctlimg.ImagesMetadata, imagesLockReader ImagesLockReader) *Bundle {
-	return &Bundle{plainimg.NewPlainImage(ref, imagesMetadata), imagesMetadata, imagesLockReader}
+	return &Bundle{plainImg: plainimg.NewPlainImage(ref, imagesMetadata), imgRetriever: imagesMetadata, imagesLockReader: imagesLockReader, imagesRef: map[string]ImageRef{}}
 }
 
 func (o *Bundle) DigestRef() string { return o.plainImg.DigestRef() }
 func (o *Bundle) Repo() string      { return o.plainImg.Repo() }
 func (o *Bundle) Tag() string       { return o.plainImg.Tag() }
 
+func (o *Bundle) AddImagesRef(refs ...ImageRef) {
```

Comment:
given all the work in this PR, im not sure i see the purpose of imageRefs on bundle being public. from what i see it's only really used by the NoteCopy.

Reply:
We are using it for testing to ensure that the found references are the ones we are expecting